From 0a5e8312806a90bb8d55b924465240476742bbb1 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Thu, 10 Aug 2017 15:26:09 +0800
Subject: [PATCH 2361/5242] MLK-16183 drm/bridge: it6263: Workaround cable
 detection failure at boot time

commit  9fac5d762e806f212e855222dde1e42ddb9545c0 from
https://source.codeaurora.org/external/imx/linux-imx.git

There is cable detection failure issue on i.MX8qxp arm2 platform
at boot time when we avoid imx-drm deferral probe entirely(i2c
bus driver probe needs to be before the it6263 driver probe).
The workaround is to read the cable detection status multiple
times.  Based on experiments, it looks reading for 40 times works.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/bridge/it6263.c |   11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/bridge/it6263.c b/drivers/gpu/drm/bridge/it6263.c
index e7aef77..9fbe764 100644
--- a/drivers/gpu/drm/bridge/it6263.c
+++ b/drivers/gpu/drm/bridge/it6263.c
@@ -345,8 +345,15 @@ static inline void hdmi_update_bits(struct it6263 *it6263, unsigned int reg,
 {
 	struct it6263 *it6263 = connector_to_it6263(connector);
 	unsigned int status;
-
-	regmap_read(it6263->hdmi_regmap, HDMI_REG_SYS_STATUS, &status);
+	int i;
+
+	/*
+	 * FIXME: We read status tens of times to workaround
+	 * cable detection failure issue at boot time on some
+	 * platforms.
+	 */
+	for (i = 0; i < 40; i++)
+		regmap_read(it6263->hdmi_regmap, HDMI_REG_SYS_STATUS, &status);
 
 	return (status & HPDETECT) ? connector_status_connected :
 					connector_status_disconnected;
-- 
1.7.9.5

